monitoring/check_collectd_mlab_test.py

```diff
@@ -0,0 +1,379 @@
+      self, mock_access):
+    # Mock out os.access to guarantee that write access is rejected.
+    mock_access.return_value = False
+
+    self.assertRaises(
+        check_collectd_mlab.ReadonlyFilesystemCriticalError,
+        check_collectd_mlab.assert_collectd_responds)
+    self.assertTrue(mock_access.called)
+
+  def testunit_assert_collectd_responds_WHEN_sock_connect_fails(self):
+    check_collectd_mlab.COLLECTD_PID = (
+        os.path.join(self._testdata_dir, 'fake_pid'))
+    check_collectd_mlab.COLLECTD_UNIXSOCK = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.SocketConnectionCriticalError,
+        check_collectd_mlab.assert_collectd_responds)
+
+  def testunit_assert_collectd_vsys_setup_WHEN_vsys_backend_is_missing(self):
+    check_collectd_mlab.VSYSPATH_BACKEND = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.MissingVsysBackendCriticalError,
+        check_collectd_mlab.assert_collectd_vsys_setup)
+
+  def testunit_assert_collectd_vsys_setup_WHEN_vsys_slice_is_missing(self):
+    check_collectd_mlab.VSYSPATH_BACKEND = os.path.join(
+        self._testdata_dir, 'fake_backend')
+    check_collectd_mlab.VSYSPATH_SLICE = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.MissingVsysFrontendCriticalError,
+        check_collectd_mlab.assert_collectd_vsys_setup)
+
+  def testunit_assert_collectd_vsys_setup_WHEN_vsys_acl_is_missing(self):
+    check_collectd_mlab.VSYSPATH_BACKEND = os.path.join(
+        self._testdata_dir, 'fake_backend')
+    check_collectd_mlab.VSYSPATH_SLICE = os.path.join(
+        self._testdata_dir, 'fake_slice')
+    check_collectd_mlab.VSYSPATH_ACL = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.MissingVsysAclCriticalError,
+        check_collectd_mlab.assert_collectd_vsys_setup)
+
+  def testunit_assert_collectd_vsys_setup_WHEN_acl_incomplete(self):
+    check_collectd_mlab.VSYSPATH_BACKEND = os.path.join(
+        self._testdata_dir, 'fake_backend')
+    check_collectd_mlab.VSYSPATH_SLICE = os.path.join(
+        self._testdata_dir, 'fake_slice')
+    check_collectd_mlab.VSYSPATH_ACL = os.path.join(
+        self._testdata_dir, 'fake_acl')
+
+    self.assertRaises(
+        check_collectd_mlab.MissingSliceFromVsysAclCriticalError,
+        check_collectd_mlab.assert_collectd_vsys_setup)
+
+  @mock.patch('subprocess.Popen')
+  def testcover_run_collectd_nagios(self, mock_popen):
+    mock_popen.return_value.wait.return_value = 2
+
+    returned_value = check_collectd_mlab.run_collectd_nagios(
+        'host', 'metric', 'value', 'warning', 'critical')
+
+    self.assertEqual(returned_value, 2)
+    self.assertTrue(mock_popen.called)
+
+  @mock.patch('check_collectd_mlab.run_collectd_nagios')
+  def testcover_assert_collectd_nagios_levels(self, mock_run_collectd_nagios):
+    # This is not ideal. But, it's just a coverage test.
+    # Non-zero values cause a failue. Cause each call to to fail in sequence.
```

Comment:
Hmm, now I have to figure out what I meant too...

Okay, so I was saying that this test is somewhat flaky because it depends on the exact number of times we call `run_collectd_nagios`, so I was thinking about a mock function that would check the metric (e.g. `meta/timer-read`) and return 1 for that specific call, 0 otherwise, but that's much more complicated and still leaves us with a pretty equally flaky test. Rescinded!